### DISCUSSION TITLE
How to redirect many pages

### DISCUSSION THREAD

**Discussion by CoreyVincent** [vuepress/core#748] [Q&A] [answered]
I'm trying to redirect a bunch of pages.  
Previously I was able to do this in the config.js:
```ts
additionalPages: [{
            path: '/from/here.html',
            filePath: path.resolve(__dirname, '../to/here.md'),
        },
        // ... many more pages
]
```
V2 suggests `app.pages.push(createPage())` in the `onInitialized` hook, but this means nothing to me.  
I'm not that smart, and the old way of doing this was easy.

I tried doing something like this, which seemed promising:
```ts
import { createPage, PageOptions } from "@vuepress/core";
const option: PageOptions = {
      path: "/from/here.html",
      filePath: "/to/here.md",
    };
    const page = await createPage(app, option);
    app.pages.push(page);
  },
};
```
And importing it into my config through the `plugins` section.  
But it doesn't route to the `filePath` location.

**Comment by Mister-Hope** [comment, marked as answer]
Your questioni is quite confusing me. What to you mean by redirect? When people are talking about redirect, (such as `a` to `b`), they probably want an unexisting `a`, and navigating to `b` automaticlly when visiting `a`.

You probably may need to describe it as cloning if you are actually providing same content in mutiple pages. Also that will make same page availble in mutiple links (i.e., `a` has the exact content comparing with `b`), but a limitation is page a and page b do not share the same state.

When you provide that config in V1, I believed you are creating a new `/from/here.html` page, which it's content are using `/to/here.md`, therefore you can visit `/from/here.html` directly and see same content of `/to/here.html`, there should be no redirects. If V1 behavior is different (such as making redirects as you describe), then I personally would regard that as a bug. 

Anyway, I think V2 is providing the right behavior. If you want a redirect, you may need to use a custom layout, providing you a blank page and redirect with the url you want later. 

---

Edited: If you are procceeding with a lot of urls, e.g.: changing a base to another, you probably need to config the server, to perform a 302 redirect with that url, it should not be set here. Provide a html blank page and redirect to another url using js should NOT be **the correct  solution**

I will mark it as solved, feel free to ask related question if you have problems achieving it with the way I mentioned.
